app/Http/Controllers/CampaignRegistrationController.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Campaign;
+use App\Models\User;
+use App\Models\Character;
+use Illuminate\Support\Facades\Auth;
+
+class CampaignRegistrationController extends Controller
+{
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $data = $request->validate([
+            'character_id' => "required|exists:characters,id",
+            'code' => "required|string"
+        ]);
+
+        $user = User::find(Auth::id())->first();
+        $character = Character::findOrFail($data['character_id'])->first();
+        $campaign = Campaign::where('code', '=', $data['code']);
+
+        if ($campaign->exists()) {
+            $campaign = $campaign->first();
+            $user->campaigns()->attach($campaign);
```

Comment:
regardless of your decision on my above comments of the request validation rules, I think you should be explicitly passing in the pivot value here as false if the user you're attaching is a player.
`$user->campaigns()->attach($campaign, ['is_dm' => false])`

Reply:
is_dm automatically defaults to false, however, I can explicitly specify it.

Reply:
ah okay i didnt know this, in that case its fine